routers/api/v1/user/app.go

```diff
@@ -300,8 +318,18 @@ func DeleteOauth2Application(ctx *context.APIContext) {
 	//     "$ref": "#/responses/empty"
 	//   "404":
 	//     "$ref": "#/responses/notFound"
-	appID := ctx.ParamsInt64(":id")
-	if err := auth_model.DeleteOAuth2Application(ctx, appID, ctx.Doer.ID); err != nil {
+
+	app, err := auth_model.GetOAuth2ApplicationByID(ctx, ctx.ParamsInt64("id"))
+	if err != nil {
+		if errors.Is(err, util.ErrNotExist) {
+			ctx.NotFound()
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetOAuth2ApplicationByID", err)
+		}
```

Comment:
There is one (`NotFoundOrServerError`) but its design is clunky. It should use the `errors.Is` check inside instead of a callback. I will not use the current version.